### ISSUE TITLE
Flaky Test: ConnectionResetMidRequestIsLoggedAsDebug

### ISSUE THREAD

**Issue by mikeharder** [aspnet/KestrelHttpServer#1766]
```
[12:12:42] :		 [Test collection for Microsoft.AspNetCore.Server.Kestrel.FunctionalTests.RequestTests (1)] ConnectionResetMidRequestIsLoggedAsDebug (1m:09s)
[12:12:42] :	 [Step 3/4]   The active test run was aborted. Reason: Unable to communicate with test host process.

ConnectionResetMidRequestIsLoggedAsDebug [FAIL]
  Assert.True() Failure
  Expected: True
  Actual:   False
  Stack Trace:
    RequestTests.cs(441,0): at Microsoft.AspNetCore.Server.Kestrel.FunctionalTests.RequestTests.<ConnectionResetMidRequestIsLoggedAsDebug>d__14.MoveNext()
    --- End of stack trace from previous location where exception was thrown ---
       at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
       at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
    --- End of stack trace from previous location where exception was thrown ---
       at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
       at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
    --- End of stack trace from previous location where exception was thrown ---
       at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
       at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
```

## Repros
4/25/17: http://aspnetci/viewLog.html?buildId=208284&buildTypeId=Lite_UniverseTest&tab=buildLog&filter=err

**Comment by davidfowl** [comment]
I'm guessing this is https://github.com/aspnet/KestrelHttpServer/issues/1761

**Comment by muratg** [comment]
@mikeharder did you see this after #1761 was resolved?

**Comment by pakrym** [comment]
Still happening a log in Windows kestrel stress test configuration

**Comment by mikeharder** [comment]
Just happened again today: http://aspnetci/viewLog.html?buildId=255643&tab=buildResultsDiv&buildTypeId=Lite_UniverseTest

**Comment by mikeharder** [comment]
This is causing a lot of failures in our 2.0 builds.  Would be great if this can be fixed before 2.0 RTM.
